src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -379,13 +379,16 @@ private int countCurrentEC2Slaves(SlaveTemplate template) throws AmazonClientExc
         Set<String> instanceIds = new HashSet<String>();
         String description = template != null ? template.description : null;
 
+        //FIXME convert to a filter query
         for (Reservation r : connect().describeInstances().getReservations()) {
             for (Instance i : r.getInstances()) {
                 if (isEc2ProvisionedAmiSlave(i.getTags(), description)
                     && isEc2ProvisionedJenkinsSlave(i.getTags(), jenkinsServerUrl)
                     && (template == null || template.getAmi().equals(i.getImageId()))) {
                     InstanceStateName stateName = InstanceStateName.fromValue(i.getState().getName());
-                    if (stateName != InstanceStateName.Terminated && stateName != InstanceStateName.ShuttingDown) {
+                    if (stateName != InstanceStateName.Terminated && 
+                        stateName != InstanceStateName.ShuttingDown && 
+                        stateName != InstanceStateName.Stopped ) {
```

Comment:
I think it'd be great to introduce some small method to clarify to intent and make it easier to reason about/understand.
Here probably what we want in the end is like

```java
if ( isInstanceRunning( i ) {
... 
}
```